runtime/gc_ctrl.c

```diff
@@ -539,7 +538,7 @@ CAMLprim value caml_gc_major(value v)
   caml_empty_minor_heap ();
   caml_finish_major_cycle ();
   test_and_compact ();
-  caml_check_urgent_gc (Val_unit);
+  caml_final_do_calls ();
```

Comment:
I would prefer no longer calling this function directly anywhere, in order to make this a general convention. Could we call `caml_process_pending_callbacks` directly instead?

Reply:
In general yes. You already break the rule in `caml_memprof_track_young` (for a good reason: you empty the queue). final_do_calls has the same property of emptying the queue, and we are in GC functions, so in doubt I left it as is. If you tell me that you are sure that caml_gc_major (and the other ones below) do not rely on this property in a subtle manner, then I can do the change. It is also possible to just change the second call to final_do_calls.

Reply:
AFAICT, I don't think anything relies on the finalisers queue being empty. It's rather a best-effort to execute them when we finish a GC cycle because that's where there will be a bunch of new ones in the queue. So yes, you can do the change.

Reply:
Done. I am not familiar enough with the GC to review your suggestion, but since it was already in reviewed code of yours, I consider it has been reviewed.